File: src/test/java/examples/java/noreceiver/doublecomparison/Incorrect0.java
package examples.java.noreceiver.doublecomparison;

public class Incorrect0 {
  public double sum(double[] values) {
    double sum = 0.0;
    for (int i = 0; i < values.length - 1; i++) {
      sum += values[i];
    }
    return sum;
  }
}

File: src/test/java/examples/java/noreceiver/kotlinsynthetic/Incorrect0.kt
package examples.java.noreceiver.kotlinsynthetic

@Suppress("UtilityClassWithPublicConstructor")
class Incorrect0 {
    companion object {
        @JvmStatic // Can't have two loose addOne Kotlin functions in the same package
        fun addOne(value: Int): Int {
            val transform = ::plus2
            return transform(value)
        }

        private fun plus2(x: Int): Int {
            return x + 2
        }
    }
}